go/engine/paperkey_gen.go

```diff
@@ -264,5 +280,31 @@ func (e *PaperKeyGen) push(ctx *Context) error {
 		Contextified:   libkb.NewContextified(e.G()),
 	}
 
-	return libkb.DelegatorAggregator(ctx.LoginContext, []libkb.Delegator{sigDel, sigEnc}, nil)
+	var sdhBoxes = []libkb.SharedDHSecretKeyBox{}
+	if e.G().Env.GetEnableSharedDH() {
+		if e.getSharedDHKeyring().CurrentGeneration() == 0 {
```

Comment:
I'd make a new little helper in `SharedDHKeyring` that just checks `Generation == 0`. Just doesn't make sense for the convention to leak out.